### PR TITLE
6893: DYN-6956 symbol nodes should not show in HomeWorkspace context.

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

relies on:
https://github.com/DynamoDS/librarie.js/pull/228

Dynamo now informs librarieJS of the workspace context, when loadedTypes that have `hiddenInWorkspaceContext == true` are rendered and the context is `home` - they are simply not rendered on the JS side.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

fix a legacy issue where input and output symbol nodes were shown in the HomeWorkspace context even though they are only useful in CustomNodeWorkspaces.

### Reviewers

(FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/LibraryViewExtensionWebView2/web/library/library.html [hunk 1]
```diff
@@ -194,6 +195,9 @@
 
         refreshLibraryView(libController);
 
+        //initial state
+        libController.setHostContext("home")
```
Comment (RobertGlobant20) [inline src/LibraryViewExtensionWebView2/web/library/library.html:199]: @Enzo707  think this line needs to be moved to librarie.js->library.html file once you consume the npm package inside Dynamo.

File: src/LibraryViewExtensionWebView2/ViewExtension.cs [hunk 2]
```diff
@@ -37,7 +38,17 @@ public void Loaded(ViewLoadedParams viewLoadedParams)
...
             }
-            
+            viewParams.CurrentWorkspaceChanged += ViewParams_CurrentWorkspaceChanged;
+        }
+
+        private void ViewParams_CurrentWorkspaceChanged(IWorkspaceModel workspace)
+        {
+            var type = workspace switch
+            {
+                HomeWorkspaceModel _ => "home",
```
Comment (RobertGlobant20) [inline src/LibraryViewExtensionWebView2/ViewExtension.cs:48]: I see you defined the next line in librarie.js
`export type HostingContextType = "home"|"custom"|"none"`
What happens if is set to "none"?

File: src/LibraryViewExtensionWebView2/LibraryViewController.cs [hunk 3]
```diff
@@ -753,5 +753,13 @@ public static async Task<string> ExecuteScriptFunctionAsync(WebView2 webView2, s
             script += ");";
             return await webView2.ExecuteScriptAsync(script);
         }
+        /// <summary>
+        /// Gives the library UI information on Dynamo's current context.
+        /// </summary>
+        /// <param name="type"></param>
+        internal void UpdateContext(string type)
+        {
+            ExecuteScriptFunctionAsync(browser,"libController.setHostContext", type);
```
Comment (RobertGlobant20) [inline src/LibraryViewExtensionWebView2/LibraryViewController.cs:762]: Usually we were defining the function in the library.html (like an interface of Dynamo and librarie.js), for example:
function SetContext(type){
     libController.setHostContext(type)
}
But I guess this way is better if now everything will be inside the librarie.js component.

**Review by RobertGlobant20 — APPROVED** [review]
LGTM

File: src/LibraryViewExtensionWebView2/ViewExtension.cs (hunk 2 above)
Comment (mjkkirschner) [reply src/LibraryViewExtensionWebView2/ViewExtension.cs:48]: custom and none do nothing special at the moment - library is rendered as before my prs.

File: src/LibraryViewExtensionWebView2/LibraryViewController.cs (hunk 3 above)
Comment (mjkkirschner) [reply src/LibraryViewExtensionWebView2/LibraryViewController.cs:762]: I actually had trouble with that pattern where `libController` was not set correctly.

File: src/LibraryViewExtensionWebView2/Handlers/NodeItemDataProvider.cs [hunk 4]
```diff
@@ -41,6 +45,7 @@ class NodeItemDataProvider : ResourceProviderBase
     {
         protected NodeSearchModel model;
         private IconResourceProvider iconProvider;
+        readonly string[] typesToHideInHomeWorkspaces = { "Core.Input.Input", "Core.Input.Output" };
```
Comment (pinzart90) [inline src/LibraryViewExtensionWebView2/Handlers/NodeItemDataProvider.cs:48]: just a fixed list here ? can we use nameof() ? or are the types not accessible in this csproj ?

File: src/LibraryViewExtensionWebView2/LibraryViewController.cs (hunk 3 above)
Comment (pinzart90) [reply src/LibraryViewExtensionWebView2/LibraryViewController.cs:762]: I don't think that we're using ExecuteScriptFunctionAsync for so many interops between c# and javascript. ExecuteScriptFunctionAsync forces you to write the javascript code inside c# (prone to errors)
There are other ways to interop (post simple  messages or hostObjects) We should review the interop approach in a separate task

File: src/LibraryViewExtensionWebView2/Handlers/NodeItemDataProvider.cs (hunk 4 above)
Comment (mjkkirschner) [reply src/LibraryViewExtensionWebView2/Handlers/NodeItemDataProvider.cs:48]: let me take a look, they should be accessible.

Comment (mjkkirschner) [reply src/LibraryViewExtensionWebView2/Handlers/NodeItemDataProvider.cs]: @pinzart90 PTAL, now using the types directly.

**Review by pinzart90 — APPROVED** [review]

**Comment by mjkkirschner** [comment]
the test failure exists on master already, merging.

**Pull request merged**